src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -666,13 +666,7 @@ private async Task<List<ChainedHeaderBlock>> RewindToForkPointAsync(ChainedHeade
 
             while (current != fork)
             {
-                await this.ConsensusRules.RewindAsync().ConfigureAwait(false);
-
-                lock (this.peerLock)
-                {
-                    this.SetConsensusTipInternalLocked(current.Previous);
-                }
-
+                // Access the block before rewinding.
```

Comment:
I agree first find the block then rewind, good fix @noescape00 ?

Reply:
Previous logic is rewind then find the block (throw if not found), new logic is find the block (throw if not found) then rewind.
Seems more correct to me.